src/python/pants/backend/python/tasks/select_interpreter.py

```diff
@@ -56,6 +56,8 @@ def product_types(cls):
     return [PythonInterpreter]
 
   def execute(self):
+    # TODO(CMLivingston): Grabbing `PythonRequirementLibrary`s here seems like it is a hack.
```

Comment:
I can take this in a follow-up, but if you understand the explanation at https://github.com/pantsbuild/pants/issues/6632#issuecomment-430593574 perhaps you can just change this TODO into a comment that points out downstream interpreter consumers may need an interpreter for any sort of importable python target (including PythonRequirementLibrary targets) even if we don't need all such targets here to setup interpreters. IE: we only need those importable targets with compatibility constraints.

Reply:
I'll add the comment.